lib/puppet/pops/evaluator/evaluator_impl.rb

```diff
@@ -232,6 +232,9 @@ def assign_Array(lvalues, values, o, scope)
           o, scope)
       end
     elsif values.is_a?(Puppet::Pops::Types::PClassType)
+      if Puppet[:tasks]
+        fail(Issues::CATALOG_OPERATION_NOT_SUPPORTED_WHEN_SCRIPTING, o, {:operation => _('multi var assignment from class')})
```

Comment:
Given https://tickets.puppetlabs.com/browse/PUP-7905, where class definitions are trapped in the validator, do we still need this?

Reply:
Yes, you can still try to write `[$a] = Class[foo]` - even if you can never define or find a class. It don't think it should be an error to use `Class` as a data type, but here it is an operation involving a class. If not treated special, an attempt would be made to look for the class in know_resource_types (which we want to get rid of).